src/Orleans.Core/Utils/KeyedService.cs

```diff
@@ -123,7 +123,7 @@ public class KeyedServiceCollection<TKey, TService> : IKeyedServiceCollection<TK
         /// <inheritdoc/>
         public TService GetService(IServiceProvider services, TKey key)
         {
-            return this.GetServices(services).FirstOrDefault(s => s.Equals(key))?.GetService(services);
+            return this.GetServices(services).LastOrDefault(s => s.Equals(key))?.GetService(services);
```

Comment:
I wonder if this change alone doesn't deserve it's own mini PR since it might change some behaviors